src/Cache.js

```diff
@@ -62,11 +62,16 @@ class Cache {
    * remove from cache after maxAge time has passed
    *
    * @param {*} key the key to remove
-   * @param {number} maxAge the time in milliseconds to wait before removing key
+   * @param {number} maxAge the time in milliseconds to wait before removing the key
+   * @param {Function} onExpiration a callback that is called after removing the key
    */
-  expireAfter(key: any, maxAge: number) {
+  expireAfter(key: any, maxAge: number, onExpiration: ?Function) {
     setTimeout(() => {
       this.remove(key);
+      if (typeof onExpiration === 'function') {
+        // We assume here that key is either an instance of one of cache key classes, or a primitive
+        onExpiration(isComplexObject(key) ? key.key : key);
```

Comment:
there is no need to perform validation here, and you are inherently limiting the possibilities here because the key can be literally anything (even `undefined`). Don't over think it,

```javascript
onExpiration(key.key);
```

The contract is guaranteed here.